src/base/utils/fs.h

```diff
@@ -71,6 +77,9 @@ namespace Utils::Fs
 
     QString tempPath();
 
+    nonstd::expected<void, QString> saveToFile(const QString &path, const QByteArray &data);
```

Comment:
I would move this helper to Utils::Misc and the second one somewhere inside BitTorrent layer (it isn't supposed to be used outside of BitTorrent layers, isn't it?).

Reply:
>I would move this helper to Utils::Misc 

I don't feel strongly. Utils::Fs has a lot helpers about file path, file name, file property. I think it fits OK.

>and the second one somewhere inside BitTorrent layer (it isn't supposed to be used outside of BitTorrent layer, isn't it?).

I don't feel strongly either. I'm concerned that the helper functions will scatter in separate files which would not be easy to see/update them all.

Reply:
>`Utils::Fs` has a lot helpers about file path, file name, file property. 

...which is basically some service/meta FS-related functions, and not regular use, such as reading/writing files.
However, I do not insist.

Reply:
>I'm concerned that the helper functions will scatter in separate files which would not be easy to see/update them all.

I would still limit the scope of libtorrent native entities within the abstraction layer as much as possible.
Or maybe there is some more abstract way?
Okay, you can leave it as it is for now.

Reply:
>...which is basically some service/meta FS-related functions, and not regular use, such as reading/writing files.

Another choice is io.cpp would it be better for you?

Reply:
>Another choice is io.cpp would it be better for you?

Yes.

Reply:
done.

Reply:
>Or maybe there is some more abstract way?

Something like:
```c++
nonstd::expected<void, QString> Utils::IO::saveToFile(const QString &path, std::function<nonstd::expected<void, QString> (Utils::IO::FileDeviceOutputIterator &)> writeFn)
{
    QSaveFile file {path};
    if (!file.open(QIODevice::WriteOnly))
        return nonstd::make_unexpected(file.errorString());

    const nonstd::expected<void, QString> writeResult = written(Utils::IO::FileDeviceOutputIterator(file));
    if (!writeResult)
        return writeResult.error();
    if (!file.flush() || !file.commit())
        return nonstd::make_unexpected(file.errorString());

    return {};
}
```